django/db/backends/utils.py

```diff
@@ -125,9 +125,10 @@ def typecast_time(s):  # does NOT store time zone information
     hour, minutes, seconds = s.split(':')
     if '.' in seconds:  # check whether seconds have a fractional part
         seconds, microseconds = seconds.split('.')
+        microseconds = microseconds.ljust(6, '0')
```

Comment:
Looks like Python's `sqlite` has faced this problem too with datetimes, their code is more elegant (https://bugs.python.org/issue14720 ).

We could use

```py
microseconds = '{:0<6.6}'.format(microseconds))
```

instead of this `ljust` line.